### DISCUSSION TITLE
Permissions by region/site?

### DISCUSSION THREAD

**Discussion by domainzero** [netbox-community/netbox#9264] [Q&A] [answered]
I'm not sure what the best way is to divide permissions based on site. I've started running into issues with limiting cable creation, module creation, etc by site or region. 

My end goal is to have two separate regions or sites and have two groups of users that can only work on either site or region.

I have been assigning permissions and using constraints, but can't find a good way to do cables, modules, etc as I can't seem to use region/site

Any help would be appreciated.

**Comment by EAleg** [comment]
Same problem...
Can you attach your example of code for permissions?

**Comment by IdRatherStand** [comment]
We have done this in our environment, for the cables you should be able to use the ```side_a_termination``` or ```side_b_termination``` properties if I remember correctly.

Module creation I would imagine is done via parent device site assignment.

**Comment by dteknet** [comment, marked as answer]
If it's still actual question, I've found out how this can be achieved:

```
[{"_termination_a_device__site__name__in": ["SITE-A", "SITE-B"]}, {"_termination_b_device__site__name__in": ["SITE-A", "SITE-B"]}]
```

**Reply by domainzero** [reply]
This was how I figured out how to do it as well, forgot to respond.  
Thanks!

**Reply by dteknet** [reply]
> This was how I figured out how to do it as well, forgot to respond. Thanks!

Maybe you have a solution for version 3.3.0? Because this "magic" is no longer working.

**Reply by domainzero** [reply]
`termination_a` and `termination_b` were removed in 3.3.0.
It's listed under breaking changes. You'll probably need to change your constraints to use the new names.

[Here's the release notes](https://github.com/netbox-community/netbox/releases/tag/v3.3.0).

**Reply by dteknet** [reply]
I'm aware of those. However, the new names listed are not available in constraints, unfortunately. 

```
Invalid filter for <class 'dcim.models.cables.Cable'>: Cannot resolve keyword 'a_terminations' into field. Choices are: _abs_length, color, created, custom_field_data, id, journal_entries, label, last_updated, length, length_unit, status, tagged_items, tags, tenant, tenant_id, terminations, type
```

**Reply by julianstolp** [reply]
I tried to restricted the cable creation to one location. This worked for me:

`{"terminations___device__location__slug__startswith": "xyz"}`

Should be adaptable to multiple sites

**Reply by dteknet** [reply]
> I tried to restricted the cable creation to one location. This worked for me:
> 
> `{"terminations___device__location__slug__startswith": "xyz"}`
> 
> Should be adaptable to multiple sites

Amazing! Thank you!
How can you explain the tripple-underscore? What does it mean?

**Reply by julianstolp** [reply]
It's listed in the code with underscore https://github.com/netbox-community/netbox/blob/ab9c253310dc36dca2aa685dc97cf9d43a518d5e/netbox/dcim/models/cables.py#L240

**Comment by nebulite** [comment]
same issues, need group by region / site  to control the devices belong to different region/site
version 3.3.9

**Reply by julianstolp** [reply]
If the devices are independent for each group you should use the tenant permissions.
